### PR TITLE
Initial commit for Read the Docs

### PR DESCRIPTION
**Pull request by henry0312**

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+}
+source_suffix = ['.rst', '.md']
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = 'LightGBM'
+copyright = '2017, Microsoft Corporation'
+author = 'Microsoft Corporation'
```
Comment (henry0312) [inline docs/conf.py]: is this ok?
I referred to https://github.com/Microsoft/LightGBM/blob/master/LICENSE

Comment (guolinke) [inline docs/conf.py]: yeah, it is ok.

**Comment by guolinke** [comment]
@henry0312 Do we need to put the build script and generated docs ?

**Comment by henry0312** [comment]
> build script

Yes, we do.
They are needed to build locally.

> generated docs

No, we don't.
Read the Docs loads `conf.py` and generates documentations.
And you need to setup webhooks to build docs automatically.
cf. http://docs.readthedocs.io/en/latest/webhooks.html#github

**Comment by guolinke** [comment]
@henry0312 OK, i will setup the hook. 
BTW, i saw a ```bat``` file, it is also can be removed ?

**Comment by henry0312** [comment]
it is needed for Windows Users, and I guess it isn't published on Read the Docs.

**Comment by guolinke** [comment]
BTW, why need to put these original docs into ```gerenal``` ?

**Comment by henry0312** [comment]
That is for ease of management.
If we don't put these into `general`, it is a bit hard to distinguish build scripts from docs.

Before:

```
docs
├── FAQ.md
├── GPU-Performance.md
├── GPU-Targets.md
├── GPU-Tutorial.md
├── GPU-Windows.md
├── Installation-Guide.md
├── Key-Events.md
├── Makefile
├── Parallel-Learning-Guide.md
├── Parameters-tuning.md
├── Parameters.md
├── Python-API.md
├── Python-intro.md
├── Quick-Start.md
├── Readme.md
├── conf.py
├── development.md
├── index.rst
├── make.bat
└── requirements.txt
```

After:

```
docs
├── Makefile
├── README.md
├── conf.py
├── general
│   ├── FAQ.md
│   ├── GPU-Performance.md
│   ├── GPU-Targets.md
│   ├── GPU-Tutorial.md
│   ├── GPU-Windows.md
│   ├── Installation-Guide.md
│   ├── Key-Events.md
│   ├── Parallel-Learning-Guide.md
│   ├── Parameters-tuning.md
│   ├── Parameters.md
│   ├── Python-API.md
│   ├── Python-intro.md
│   ├── Quick-Start.md
│   ├── Readme.md
│   └── development.md
├── index.rst
├── make.bat
└── requirements.txt
```

And then, we will put `python`, `R`, and `cli` if needed.

**Comment by guolinke** [comment]
@henry0312 However, move them will cause many link broken ...

**Comment by henry0312** [comment]
Oh... 😢
I'll put them back.

**Comment by henry0312** [comment]
I've finished initial setup for Read the Docs.
I'm going to create actual documentation for Read the Docs.

**Pull request merged**